File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        //Optimal solution 
        //Using Dutch National flag
        int n = nums.size();
        int low = 0, mid = 0, high = n-1;
        while(mid<=high){
            if(nums[mid] == 0){
                swap(nums[low], nums[mid]);
                low++;
                mid++;
            }else if(nums[mid] == 1){
                mid++;
            }else{
                swap(nums[high], nums[mid]);
                high--;
            }
        }
    
    }
};

File: 0075-sort-colors/0075-sort-colors.cs
public class Solution {
    public void SortColors(int[] nums) {
        int zerosCount = 0, onesCount = 0, twosCount = 0;
        for(int i = 0;i<nums.Length;i++){
            if(nums[i] == 0) zerosCount++;
            else if(nums[i] == 1) onesCount++;
            else twosCount++; 
        }
        for(int i = 0;i<nums.Length;i++){
            if(zerosCount > 0){
                nums[i] = 0;
                zerosCount--;
            }else if(onesCount > 0){
                nums[i] = 1;
                onesCount--;
            }else{
                nums[i] = 2;
                twosCount--;
            }
        }
    }
}